### PR TITLE
Lung hot air damage change

### PR DESCRIPTION
**Pull request by vode-code**
## About the PR 
Makes lungs take damage based on their temp_tolerance, instead of a single hardcoded number.
Also makes them take damage separately, except for the damage caused by the affected lung's TakeDamage.

## Why's this needed? 
Cyberlungs are currently only able to deal with hot air damage by nullifying it, and will take maximum damage as soon as they are able to take damage at all.
This fixes that.

## Changelog

```
(u)Chickenish:
(+)Cyberlungs now have improved resistance to hot air, reducing damage even past the temperature threshold.
(+)Singular lungs now have more intuitive interactions with hot air.
```

### CODE REVIEW

File: code/mob/living/carbon/human/procs/Life.dm [hunk 1]
```diff
@@ -933,13 +933,27 @@
 
 		if (breath.temperature > min(organHolder.left_lung ? organHolder.left_lung.temp_tolerance : INFINITY, organHolder.right_lung ? organHolder.right_lung.temp_tolerance : INFINITY) && !src.is_heat_resistant()) // Hot air hurts :(
 			//checks the temperature threshold for each lung, ignoring missing ones. the case of having no lungs is handled in handle_breath.
-			var/burn_damage = min((breath.temperature - (T0C+66)) / 3,10) + 6
-			TakeDamage("chest", 0, burn_damage, 0, DAMAGE_BURN)
+			var/lung_burn_left = min(max(breath.temperature - organHolder.left_lung?.temp_tolerance, 0), 10)
+			var/lung_burn_right = min(max(breath.temperature - organHolder.right_lung?.temp_tolerance, 0), 10)
+//			var/burn_damage = min((breath.temperature - (T0C+66)) / 3,10) + 6
```
Comment (ZeWaka) [inline code/mob/living/carbon/human/procs/Life.dm:938]: delete this

File: code/mob/living/carbon/human/procs/Life.dm [hunk 2]
```diff
@@ -933,13 +933,27 @@
...
+/*		TakeDamage("chest", 0, burn_damage, 0, DAMAGE_BURN)
 			if (prob(20))
-				boutput(src, "<span class='alert'>You feel a searing heat in your lungs!</span>")
-				if (src.organHolder)
-					src.organHolder.damage_organs(0, max(burn_damage, 3), 0, list("left_lung", "right_lung"), 80)
-
+			boutput(src, "<span class='alert'>You feel a searing heat in your lungs!</span>")
+			if (src.organHolder)
+				src.organHolder.damage_organs(0, max(burn_damage, 3), 0, list("left_lung", "right_lung"), 80)
+*/
```
Comment (ZeWaka) [inline code/mob/living/carbon/human/procs/Life.dm:956]: delete this don't just comment it out

File: code/mob/living/carbon/human/procs/Life.dm (hunk 1 above)
Comment (vode-code) [reply code/mob/living/carbon/human/procs/Life.dm]: that was what you were asking for, right?

File: code/mob/living/carbon/human/procs/Life.dm [hunk 3]
```diff
@@ -933,12 +933,20 @@
...
 			//checks the temperature threshold for each lung, ignoring missing ones. the case of having no lungs is handled in handle_breath.
-			var/burn_damage = min((breath.temperature - (T0C+66)) / 3,10) + 6
-			TakeDamage("chest", 0, burn_damage, 0, DAMAGE_BURN)
-			if (prob(20))
-				boutput(src, "<span class='alert'>You feel a searing heat in your lungs!</span>")
-				if (src.organHolder)
-					src.organHolder.damage_organs(0, max(burn_damage, 3), 0, list("left_lung", "right_lung"), 80)
+			var/lung_burn_left = min(max(breath.temperature - organHolder.left_lung?.temp_tolerance, 0) / 3, 10)
+			var/lung_burn_right = min(max(breath.temperature - organHolder.right_lung?.temp_tolerance, 0) / 3, 10)
+			if (breath.temperature > organHolder.left_lung ? organHolder.left_lung.temp_tolerance : INFINITY)
```
Comment (ZeWaka) [inline code/mob/living/carbon/human/procs/Life.dm:938]:
```suggestion
			if (breath.temperature > (organHolder.left_lung ? organHolder.left_lung.temp_tolerance : INFINITY))
```

File: code/mob/living/carbon/human/procs/Life.dm [hunk 4]
```diff
@@ -933,12 +933,20 @@
...
+				TakeDamage("chest", 0, (lung_burn_left / 2) + 3, 0, DAMAGE_BURN)
+				if(prob(20))
+					boutput(src, "<span class='alert'>This air is searing hot!</span>")
+					if (prob(80))
+						src.organHolder.damage_organ(0, lung_burn_left + 6, 0, "left_lung")
+			if (breath.temperature > organHolder.right_lung ? organHolder.right_lung.temp_tolerance : INFINITY)
```
Comment (ZeWaka) [inline code/mob/living/carbon/human/procs/Life.dm:944]:
```suggestion
			if (breath.temperature > (organHolder.right_lung ? organHolder.right_lung.temp_tolerance : INFINITY))
```

**Pull request merged**